lib/src/facts/bsd/networking_resolver.cc

```diff
@@ -94,21 +95,12 @@ namespace facter { namespace facts { namespace bsd {
 
     void networking_resolver::populate_address(interface& iface, ifaddrs const* addr) const
     {
-        string* address = nullptr;
-        if (addr->ifa_addr->sa_family == AF_INET) {
-            address = &iface.address.v4;
-        } else if (addr->ifa_addr->sa_family == AF_INET6) {
-            address = &iface.address.v6;
-        } else if (is_link_address(addr->ifa_addr)) {
-            address = &iface.macaddress;
-        }
-
-        if (!address) {
-            // Unsupported address
+        if (is_link_address(addr->ifa_addr)) {
+            iface.macaddress = address_to_string(addr->ifa_addr);
             return;
         }
 
-        *address = address_to_string(addr->ifa_addr);
+        assign_address(iface.address, addr->ifa_addr);
```

Comment:
So we appear to expect a method named assign_address to be defined? Should we add that as an abstract method to the bsd::networking_resolver?